back-end/src/sciler/communication/communication.go

```diff
@@ -11,17 +10,51 @@ import (
 
 // Communicator is a type that maintains communication with the front-end and the client computers.
 type Communicator struct {
-	client           mqtt.Client
-	topicsOfInterest []string
+	client mqtt.Client
 }
 
-// NewCommunicator is a constructor for a Communicator
-func NewCommunicator(host string, port int, topicsOfInterest []string) *Communicator {
+// NewCommunicator is a constructor that sets up a communicator
+// host string host address of the broker
+// port int mqtt port of the broker
+// topicsOfInterest []string all topic to which to subscribe to
+// handler function(Client, Message) function that handles all incoming messages
+// onStart function() function that will be performed on startup
+func NewCommunicator(host string, port int, topicsOfInterest []string, handler mqtt.MessageHandler, onStart func()) *Communicator {
 	opts := mqtt.NewClientOptions()
 	opts.AddBroker(fmt.Sprintf("%s://%s:%d", "tcp", host, port))
 	opts.SetClientID("back-end")
-	opts.SetConnectionLostHandler(onConnectionLost)
-	opts.SetKeepAlive(20)
+	opts.SetConnectionLostHandler(func(client mqtt.Client, err error) {
+		if err.Error() == "EOF" {
+			logger.Warnf("connection lost: broker closed connection, (multiple client ID: %s?)", opts.ClientID)
+		} else {
+			logger.Warnf("connection lost: %v", err)
+		}
+	})
+	opts.SetConnectRetry(false)
+	opts.SetAutoReconnect(true)
+
+	timeout := time.Duration(5) * time.Second
```

Comment:
why 5? before we had 20? should they really all be 5?

Reply:
5 is faster than 20, (and still stable), 
When the back-end disconnect for any reason, we want to reconnect asap

Reply:
but the timeout is also shortened

Reply:
yes, see comments to see what each timeout does

Reply:
maybe only the connect time out should be 20 again, since we want longer than 5 seconds to keep trying

Reply:
Done